Mlem/Views/Shared/CollapsibleSection.swift

```diff
@@ -0,0 +1,69 @@
+//
+//  CollapsibleSection.swift
+//  Mlem
+//
+//  Created by Sjmarf on 02/01/2024.
+//
+
+import SwiftUI
+
+struct CollapsibleSection<Content: View>: View {
+    var header: String?
+    var footer: String?
+
+    @ViewBuilder var content: () -> Content
+    @State private var collapsed: Bool
+  
+    init(
+        _ header: String? = nil,
+        footer: String? = nil,
+        collapsed: Bool = false,
+        @ViewBuilder _ content: @escaping () -> Content
+    ) {
+        self.header = header
+        self.footer = footer
+        self.content = content
+        self._collapsed = State(wrappedValue: collapsed)
+    }
+    
+    var body: some View {
+        VStack(alignment: .leading, spacing: 0) {
+            if let header {
+                HStack {
+                    Text(header)
+                        .textCase(.uppercase)
+                        .opacity(0.5)
+                    Spacer()
+                    Image(systemName: Icons.dropdown)
+                        .fontWeight(.semibold)
+                        .foregroundStyle(Color.accentColor)
+                        .rotationEffect(Angle(degrees: collapsed ? -90 : 0))
+                }
+                .font(.footnote)
+                .padding(.vertical, 6)
+                .padding(.horizontal, 16)
+                .onTapGesture { withAnimation(.default) { collapsed.toggle() }}
```

Comment:
I think it feels a little better if the whitespace is tappable too

```suggestion
                .contentShape(.rect)
                .onTapGesture { withAnimation(.default) { collapsed.toggle() }}
                .padding(.vertical, 6)
                .padding(.horizontal, 16)
```